### PR TITLE
APEX-1637 - Manual testing

### PR DESCRIPTION
**Pull request by BaDo2001**
- Add manual testing for existing features
- Fix sending schedule and cron events to staging default eventbus
- Some refactors

### CODE REVIEW

File: packages/core/db/queries/getUser.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { eq } from "drizzle-orm";
+
+import { db } from "@/db/index";
+import { users } from "@/db/schema";
+
+export const getUser = async (userId: string) =>
```
Comment (debrecenid) [inline packages/core/db/queries/getUser.ts:6]: The deleUser uses the teamId too, not seems consistent, and if including the teamId too, then we have a getTeammates which are doing the same with a bit differnt sytax.

File: stacks/utils/getFunctionProps.ts [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
-import { EventBusStack } from "./EventBusStack";
-import { SchedulerStack } from "./SchedulerStack";
-import { StorageStack } from "./StorageStack";
+import { ConfigStack } from "../ConfigStack";
+import { EventBusStack } from "../EventBusStack";
+import { SchedulerStack } from "../SchedulerStack";
+import { StorageStack } from "../StorageStack";
 
-export const getFunctionProps = () => {
+export const getBaseFunctionProps = () => {
```
Comment (debrecenid) [inline stacks/utils/getFunctionProps.ts]: Overall im not sure about that functionProps util at all, not sure that is truly the base one, the migrationDb does not need the eventbridge stuff(bind,permission) and also the scheduler lambda arns is not needed for the migration i guess.

File: stacks/utils/addTestRoutes.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import type { Api, Stack } from "sst/constructs";
+
+export const addTestRoutes = (stack: Stack, api: Api) => {
+  if (stack.stage === "production") {
```
Comment (debrecenid) [inline stacks/utils/addTestRoutes.ts:4]: Like that solution. 🎖

**Review by debrecenid — CHANGES_REQUESTED** [review]

File: packages/functions/utils/lambda/cronHandler.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ type Event =
 
 const isApiGatewayProxyEventV2 = (
   event: Event,
-): event is APIGatewayProxyEventV2 => "queryStringParameters" in event;
+): event is APIGatewayProxyEventV2 => "routeKey" in event;
```
Comment (debrecenid) [inline packages/functions/utils/lambda/cronHandler.ts:14]: Why is this change needed? What does it do?

File: packages/core/db/queries/getUser.ts (hunk 1 above)
Comment (BaDo2001) [reply packages/core/db/queries/getUser.ts:6]: Ur right but this query is only used for testing and because we are only going to use it inhouse anyway, the team id would be the same in each query.

I am happy to add the teamId in, but then the query params need to include it as well and it seems redundant as this is just for testing.

getTeammates is different because it gets everyone except that user so its not applicable.

File: packages/functions/utils/lambda/cronHandler.ts (hunk 4 above)
Comment (BaDo2001) [reply packages/functions/utils/lambda/cronHandler.ts:14]: So this util wraps a function so that it is able to accept a cron event both from event bridge and lamdba. But I need a way to distinguish as I need the eventId.

Initially I used queryStringParameters but now I am sometimes calling lambdas without any query params and in that case this check would fail and the logic would falsely identify a lambda event as an eventbridge event.

File: packages/core/db/queries/getUser.ts (hunk 1 above)
Comment (debrecenid) [reply packages/core/db/queries/getUser.ts:6]: I'm okay to leave it; just my eyes stopped there; why are we leaving the team here and not the other places? Actually it's not needed at all atm. But that's makes sense don't need to complicate the endpoint with additional params

You are right about the getTeammates. That's my bad; there is a not(..

File: packages/functions/utils/lambda/cronHandler.ts (hunk 4 above)
Comment (debrecenid) [reply packages/functions/utils/lambda/cronHandler.ts:14]: I see, that's a nice one.

File: stacks/utils/getFunctionProps.ts (hunk 2 above)
Comment (BaDo2001) [reply stacks/utils/getFunctionProps.ts]: I created 4 groups, the base one is just the secrets and runtime, there's one for the db (used by migration fn), one for eventbus (used by scheduleHandler) and one which includes everything

**Review by debrecenid — COMMENTED** [review]
The changes are looking good to me. I will approve after the failing tests have been fixed.

File: packages/functions/lambdas/manualBotJoinedEvent.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    return {
+      statusCode: 500,
+      body: JSON.stringify(
+        {
+          error,
+        },
+        null,
+        2,
+      ),
```
Comment (attilazsoltdanku) [inline packages/functions/lambdas/manualBotJoinedEvent.ts]: I guess It will be nicer to put out a util function, we use this error message handler in multiple files.

Comment (BaDo2001) [reply packages/functions/lambdas/manualBotJoinedEvent.ts]: Created utils for both okResult and errorResult

**Review by debrecenid — APPROVED** [review]

**Review by debrecenid — APPROVED** [review]

**Review by attilazsoltdanku — APPROVED** [review]
